### DISCUSSION TITLE
Provide Storybook entry-points (and why not easy installation from the CLI)

### DISCUSSION THREAD

**Discussion by jclaveau** [quasarframework/quasar#16037] [Ideas / Proposals]
[Storybook](storybook.js.org/) is a great tool, especially with [Chromatic](https://www.chromatic.com), to help people interact about the UX question.

If it is heavier than [Histoire](https://histoire.dev/), it also has more features.

I just ended the configuration of `Storybook 7` for `Quasar/Vite` with `Sass` and `Chromatic` support: https://github.com/quasarframework/quasar/issues/11654#issuecomment-1618295838

A big part of this code should belong to Quasar's code-base, in my humble opinion and (almost everytime I need to import from Quasar's file). Providing a `viteFinal()` and a `.quasar/client-entry-storybook.js` to use respectivelly in `main.ts` and `preview.ts` would be perfect.

Also, a wizard during the `create quasar` script would be super nice but purely optional.

Hoping this would be enough valuable to be considered,
Thank you very much for the quality of your work

Some related discussions:
- https://github.com/quasarframework/quasar/discussions/13147
- https://github.com/quasarframework/quasar/discussions/15585
- https://github.com/quasarframework/quasar/discussions/11537

**Comment by IlCallo** [comment]
Hey there, would you be interested into writing a guide about how to integrate it with a Quasar project?
Writing an App Extension for streamline its usage would be even better! 😁

We can provide some guidance about how to write an app extension, but we don't have enough bandwidth to implement and maintain the AE ourselves

**Reply by jclaveau** [reply]
I'm presently in a hurry for my current project (using Quasar) but if it continues after the current deadline, I would take the time for implementing this extension with pleasure (Maybe on next fall but more probably next winter).

However, there are still some parts that will probably impact the Quasar code base like the Esbuild patch (support for require()) and the requirement for Vite 4 (app-vite will be updated soon but the current i18n vite plugin is archived and has moved to `@intlify/unplugin-vue-i18n`).

Are the AEs committed to the Quasar Repo or on separated ones?

Also, I would definitely need a fine review concerning the Vite configs merge (what is really required and what is not).

Anyway, thank you for your interest and fast answer

**Reply by IlCallo** [reply]
You can find all Quasar app extensions via this link: https://www.npmjs.com/search?q=quasar-app-extension
They're all into separate repos, even the first-party ones

Vite 4 will be supported into the next app-vite major version, which will be soon released

**Reply by jclaveau** [reply]
I'll give a look to the app extension creation process a soon I have room for that. Thank you
